lib/assembly/assembly.py

```diff
@@ -181,6 +181,34 @@ def arast_reads(filelist):
         filedicts.append({'type':'single', 'files':[f]})
     return filedicts
 
+def is_long_read_file(file):
+    if re.search(r'\.h5$', file, re.IGNORECASE) is not None:
+        return True
+
+    handle = None
+    ftype = None
+    if re.search(r'\.fa$|\.fasta$', file, re.IGNORECASE) is not None:
+        handle = open(file, "rU")
+        ftype = 'fasta'
+    elif re.search(r'\.fq$|\.fastq$', file, re.IGNORECASE) is not None:
+        handle = open(file, "rU")
+        ftype = 'fastq'
+
+    seen_long = False
+    sample = 50
+    thresh = 350
+    if handle is not None:
+        for i, record in enumerate(SeqIO.parse(handle, ftype)):
```

Comment:
Rather than C style iterating, might be cleaner to slice the generator with itertools (need to import).  Also we can use the `with` statement to handle file context.  Here's a version :
```
def is_long_read_file(file, sample=50, thresh=350):
    if re.search(r'\.h5$', file, re.IGNORECASE) is not None:
        return True
    if re.search(r'\.fa$|\.fasta$', file, re.IGNORECASE) is not None:
        ftype = 'fasta'
    elif re.search(r'\.fq$|\.fastq$', file, re.IGNORECASE) is not None:
        ftype = 'fastq'
    else:
        return False

    with open(file, 'rU') as handle:
        for record in itertools.islice(SeqIO.parse(handle, ftype), 0, sample):
            if len(record.seq) > thresh:
                logger.info("Long read detected: {} = {} bp".format(record.id, len(record.seq)))
                return True
        return False

```